dotnet/src/Connectors/Connectors.Memory.PgVector/PostgresCollectionOptions.cs

```diff
@@ -10,6 +10,20 @@ namespace Microsoft.SemanticKernel.Connectors.PgVector;
 /// </summary>
 public sealed class PostgresCollectionOptions
 {
+    /// <summary>
+    /// Initializes a new instance of the <see cref="PostgresCollectionOptions"/> class.
+    /// </summary>
+    public PostgresCollectionOptions()
+    {
+    }
+
+    internal PostgresCollectionOptions(PostgresCollectionOptions? source, IEmbeddingGenerator embeddingGenerator)
```

Comment:
we have the copy ctor just to make it more likely to not miss extending the copy logic when a new property gets introduced in the future

Reply:
nit: it's internal so it doesn't matter much, but I'd just have this a simple copy constructor - accepting only another PostgresCollectionOptions instance - and then set the embedding generator as usual at the call site:

```c#
var newOptions = new PostgresCollectionOptions(source)
{
    EmbeddingGenerator = embeddingGenerator;
}
```

This would make this usable for other needs and just be more standard (same with the other option constructors).